Fails to save mapping file if directory doesn't exist

```
javax.xml.bind.JAXBException: null
	at javax.xml.bind.helpers.AbstractMarshallerImpl.marshal(AbstractMarshallerImpl.java:108) ~[na:1.8.0_144]
	at io.atlasmap.service.AtlasService.saveMappingToFile(AtlasService.java:414) [atlas-service-1.21.0-SNAPSHOT.jar:1.21.0-SNAPSHOT]
	at io.atlasmap.service.AtlasService.saveMapping(AtlasService.java:374) [atlas-service-1.21.0-SNAPSHOT.jar:1.21.0-SNAPSHOT]
	at io.atlasmap.service.AtlasService.createMappingRequest(AtlasService.java:255) [atlas-service-1.21.0-SNAPSHOT.jar:1.21.0-SNAPSHOT]
    ....
Caused by: java.io.FileNotFoundException: target/mappings/atlasmapping-UI.160487.xml (No such file or directory)
	at java.io.FileOutputStream.open0(Native Method) ~[na:1.8.0_144]
	at java.io.FileOutputStream.open(FileOutputStream.java:270) ~[na:1.8.0_144]
	at java.io.FileOutputStream.<init>(FileOutputStream.java:213) ~[na:1.8.0_144]
	at java.io.FileOutputStream.<init>(FileOutputStream.java:162) ~[na:1.8.0_144]
	at javax.xml.bind.helpers.AbstractMarshallerImpl.marshal(AbstractMarshallerImpl.java:101) ~[na:1.8.0_144]
	... 74 common frames omitted
```
And the directory is just a fixed constant... should be created if not exist

--- a/atlas-service-parent/atlas-service/src/main/java/io/atlasmap/service/AtlasService.java
+++ b/atlas-service-parent/atlas-service/src/main/java/io/atlasmap/service/AtlasService.java
@@ -411,8 +411,17 @@ public class AtlasService extends Application {
         marshaller = jaxbContext.createMarshaller();
         marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, true);
 
-        marshaller.marshal(atlasMapping,
-                new File(baseFolder + File.separator + generateMappingFileName(atlasMapping.getName())));
+        marshaller.marshal(atlasMapping, createMappingFile(atlasMapping.getName()));
+    }
+
+    private File createMappingFile(String mappingName) {
+        File dir = new File(baseFolder);
+        if (!dir.exists()) {
+            dir.mkdirs();
+        }
+        String fileName = baseFolder + File.separator + generateMappingFileName(mappingName);
+        logger.debug("Creating mapping file '{}'", fileName);
+        return new File(fileName);
     }
 
     protected String generateMappingFileName(String mappingName) {

